pennylane/ops/op_math/adjoint_class.py

```diff
@@ -224,3 +232,20 @@ def diagonalizing_gates(self):
 
     def adjoint(self):
         return self.base.queue()
+
+    @property
+    def arithmetic_depth(self) -> int:
+        return 1 + self.base.arithmetic_depth
+
+    def simplify(self, depth=-1):
+        # TODO: Should we add a self.do_queue attribute and instantiate the simplified class
+        # with the same value?
+        if depth == 0:
+            return self
+        if isinstance(self.base, Adjoint) and (depth > 1 or depth < 0):  # Adj(Adj(A)) = A
```

Comment:
```suggestion
        if isinstance(self.base, Adjoint) and (depth > 1):  # Adj(Adj(A)) = A
```
The nice thing about having `-1` mean "keep expanding" is that we can keep subtracting from it and stay in the `depth<0` keep expanding behaviour.

Reply:
I don't think we can remove this check right now. The `depth > 1` is needed because the depth is reduced by 2 (unless we change the meaning of the `depth` argument). If we remove the `depth < 0` check, then:
```python
qml.adjoint(qml.adjoint(qml.PauliX(0))).simplify() = qml.adjoint(qml.adjoint(qml.PauliX(0)))
```

Reply:
Depth has been removed from simplify.